From 3f947faca0fb91f4b1104862f7c71bf3f6bd5d48 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Fri, 7 Aug 2015 15:24:38 +0200
Subject: [PATCH 1/2] serial: reset state at startup

Message-id: <1438961078-13625-1-git-send-email-pbonzini@redhat.com>
Patchwork-id: 67393
O-Subject: [RHEL7.2 qemu-kvm-rhev PATCH] serial: reset state at startup
Bugzilla: 922014
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>
RH-Acked-by: Andrew Jones <drjones@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>

When a serial port is started, its initial state is all zero.  Make
it consistent with reset state instead.

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
Upstream commit 4df7961faaa317d57e873ecdec58422d3f979336

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/char/serial.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/hw/char/serial.c b/hw/char/serial.c
index 0c472e2..2383c31 100644
--- a/hw/char/serial.c
+++ b/hw/char/serial.c
@@ -685,6 +685,7 @@ void serial_init_core(SerialState *s)
 
     qemu_chr_add_handlers(s->chr, serial_can_receive1, serial_receive1,
                           serial_event, s);
+    serial_reset(s);
 }
 
 void serial_exit_core(SerialState *s)
-- 
1.8.3.1

